elm/Helpers/Events.elm

```diff
@@ -1,9 +1,21 @@
-module Helpers.Events exposing (onChange)
+module Helpers.Events exposing (onChange, onDeleteEmpty)
 
 import Html exposing (Attribute)
-import Html.Events exposing (on, targetValue)
-import Json.Decode exposing (map)
+import Html.Events exposing (on, targetValue, keyCode)
+import Json.Decode as Json exposing (map)
 
 onChange : (String -> msg) -> Attribute msg
 onChange tagger =
-  on "change" (map tagger targetValue)
+  on "change" <| map tagger targetValue
+
+onDeleteEmpty : String -> msg -> Attribute msg
+onDeleteEmpty str m =
+    let
+        checker : Int -> Json.Decoder msg
+        checker code = case (code == 8, str == "") of
+            (True, True) -> Json.succeed m
+            (True, False) -> Json.fail "Text not empty"
+            (False, True) -> Json.fail "Wrong key pressed"
+            (False, False) -> Json.fail "Wrong key pressed, text not empay"
```

Comment:
Typo: "empay"

Reply:
to be fair the text probably also isn't "empay"

Reply:
although I suppose I could make that assertion more explicit